### PR TITLE
Fix Issue#24491

### PR DESCRIPTION
**Pull request by ejwessel**
## **Description**

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/24535?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/24491

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by sleepytanya** [comment]
Works great! Non-numerical chars can't be typed into the Amount field.

File: ui/components/ui/unit-input/unit-input.component.js [hunk 1]
```diff
@@ -179,7 +179,16 @@ export default class UnitInput extends PureComponent {
...
+                  e.key === 'Tab' ||
+                  e.key === 'ArrowLeft' ||
+                  e.key === 'ArrowRight' ||
+                  e.key === 'Delete'
+
+                // Allow numeric input and decimal point
+                if (!isNumericInput && !isControlKey) {
                   e.preventDefault();
                 }
               }}
```
Comment (BZahory) [inline ui/components/ui/unit-input/unit-input.component.js:194]: Can we move this to a handler similar to what we do with the other attributes (e.g., `onBlur`, `onChange`)?

File: ui/components/ui/unit-input/unit-input.component.js [hunk 2]
```diff
@@ -106,6 +106,21 @@ export default class UnitInput extends PureComponent {
     this.props.onChange(value);
   };
 
+  handleOnKeyDown = (e) => {
+    const isNumericInput = DECIMAL_INPUT_REGEX.test(e.key)
```
Comment (micaelae) [inline ui/components/ui/unit-input/unit-input.component.js:110]: Can `this.keyPressRegex` be used here instead?

Comment (ejwessel) [reply ui/components/ui/unit-input/unit-input.component.js:110]: it does not unfortunately

**Review by micaelae — APPROVED** [review]

**Review by BZahory — APPROVED** [review]

**Comment by ejwessel** [comment]
https://github.com/MetaMask/metamask-extension/issues/24491

**Pull request merged**